File: ch2/scope/main.go
package main

import "fmt"

var a = "hello"

func main() {
	fmt.Printf("global a is %v, address: %v \n", a, &a)
	a := "hello"
	fmt.Printf("main a is %v, address: %v \n", a, &a)
	for i := 1; i < 2; i++ {
		a := "hello"
		fmt.Printf("for a is %v, address: %v \n", a, &a)
		if i%2 != 0 {
			a := "hello"
			fmt.Printf("if a is %v, address: %v \n", a, &a)
		}

	}

}

/**
执行结果  说明go里面不同作用域的变量,地址不一样,重新生成了对象
global a is hello, address: 0x5a60c0
main a is hello, address: 0xc000088230
for a is hello, address: 0xc000088250
if a is hello, address: 0xc000088270
*/

File: ch2/scope/main.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-
a = "hello"


def main():
    global a
    print(f"global a is {a}, address: {id(a)}")
    a = "hello1"
    print(f"local a is {a}, address: {id(a)}")
    for i in range(2):
        a = "hello2"
        print(f"for a is {a}, address: {id(a)}")
        if i % 2 != 0:
            a = "hello3"
            print(f"if a is {a}, address: {id(a)}")


if __name__ == "__main__":
    main()



"""
执行结果 Python
global a is hello, address: 2056553020144
local a is hello1, address: 2056553020912
for a is hello2, address: 2056554223984
for a is hello2, address: 2056554223984
if a is hello3, address: 2056554224304

"""